### ISSUE TITLE
"checking for ImageMagick version >= 6.4.9... no" on Debian Sid + rbenv

### ISSUE THREAD

**Issue by btakita** [gemhome/rmagick#161]
Hello, I created an issue https://github.com/rmagick/rmagick/issues/126 before I found out about this fork.

**Comment by bf4** [comment]
Thanks and sorry about that.  We can't get rmagick/rmagick back :(

**Comment by bf4** [comment]
Can you summarize, here, what your problem was and, it appears, your proposed solution? Or any thing unclear in the docs?

**Comment by btakita** [comment]
I was not able to install rmagick on Debian sid due to extconf.rb failing
to detect both ImageMagick & wand/MagickWand.h

Also, there was an issue with `require 'rmagick/version'` in `extconf.rb`.

The pull request fixes the install.

**Comment by u338steven** [comment]
> I was not able to install rmagick on Debian sid due to extconf.rb failing
to detect both ImageMagick & wand/MagickWand.h

rmagick/rmagick#126
> I have ```graphicsmagick-libmagick-dev-compat``` installed.

RMagick does not support GraphicsMagick now.

ChangeLog:
> RMagick 2.0.0
> ...
>    o Removed support for GraphicsMagick. As a result these methods are no
>      longer available: Image#grayscale_pseudo_class, Image#statistics.

If your system has GraphicsMagick and ImageMagick,
ImageMagick's paths should have priority over GraphicsMagick's so as to build RMagick.

On your box, GraphicsMagick's path might take priority.

rmagick/rmagick#126
> Here's the result of running convert.
>
> $ convert -version
> GraphicsMagick 1.3.20 2014-08-16 Q8 http://www.GraphicsMagick.org/

 ---

I guess this is solution:

step 1. make a path to ImageMagick's ```convert```
e.g.
```sh
export PATH=/usr/local/bin/:$PATH
```

step 2. gem install with ImageMagick's paths
e.g.
```sh
gem install rmagick --with-opt-include=/usr/include/ImageMagick-6/
```
(perhaps ```--with-opt-lib```, ```--without-opt-include``` or ```--without-opt-lib``` might be needed)

Or uninstall graphicsmagick-libmagick-dev-compat & reinstall ImageMagick.
 ---

> Also, there was an issue with `require 'rmagick/version'` in `extconf.rb`.

+1

**Comment by btakita** [comment]
`rmagick` should be able to install even with `graphicsmagick-libmagick-dev-compat` on the system.

I could only find `convert` with a graphicsmagick package. `pkg-config` seems to do the trick.

**Comment by u338steven** [comment]
Oh... I might understand, let me get this straight.
Is this correct?

 ---

Correctly:
You tried to build RMagick with ImageMagick, and you also have GraphicsMagick and ```graphicsmagick-libmagick-dev-compat```.

Not: (I thought this...)
You tried to build RMagick with GraphicsMagick and ```graphicsmagick-libmagick-dev-compat```.
not build with ImageMagick, but you merely have ImageMagick.

 ---

Issues:
1. Build failed at checking version, because GraphicsMagick ```convert``` was called
2. Missing ```MagickWand.h```, because GraphicsMagick ```Magick-config``` was called
3. Wrong require path ```rmagick/version```

 ---

Your solutions:
for Issue 1. Skip cheking version if ```pkg-config``` is used
for Issue 2. Give priority to ```pkg-config``` over ```Magick-config```
for Issue 3. Fix path

**Comment by btakita** [comment]
I just created a pull request for this repo. Sorry, I lost track of this. This fixes the issue in code, instead of documentation.

**Comment by btakita** [comment]
ping

**Comment by btakita** [comment]
I'm able to install the `master` branch on `gemhome/rmagick`, so I'm closing this issue.
